internal/cli/atlas/backup/compliancepolicy/copyprotection.go

```diff
@@ -0,0 +1,117 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package compliancepolicy
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+	atlasv2 "go.mongodb.org/atlas-sdk/v20230201003/admin"
+)
+
+type CopyProtectionOpts struct {
+	cli.GlobalOpts
+	cli.WatchOpts
+	policy *atlasv2.DataProtectionSettings
+	store  store.CompliancePolicy
+	enable bool
+}
+
+const (
+	enable  string = "enable"
+	disable string = "disable"
+	active  string = "ACTIVE"
+)
+
+func (opts *CopyProtectionOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+func (opts *CopyProtectionOpts) PreRun() error {
+	currentPolicy, err := opts.store.DescribeCompliancePolicy(opts.ConfigProjectID())
+	if err != nil {
+		return err
+	}
+	opts.policy = currentPolicy
+	return nil
+}
+
+func (opts *CopyProtectionOpts) copyProtectionWatcher() (bool, error) {
+	res, err := opts.store.DescribeCompliancePolicy(opts.ConfigProjectID())
+	if err != nil {
+		return false, err
+	}
+	if res.GetState() == "" {
+		return false, fmt.Errorf("could not access State field")
+	}
+	return (res.GetState() == active), nil
+}
+
+func (opts *CopyProtectionOpts) Run() error {
+	opts.policy.SetCopyProtectionEnabled(opts.enable)
+	_, err := opts.store.UpdateCompliancePolicy(opts.ConfigProjectID(), opts.policy)
+	if err != nil {
+		return err
+	}
+	if err := opts.Watch(opts.copyProtectionWatcher); err != nil {
+		return err
+	}
+
+	fmt.Printf("Your copy protection has been set to: %v\n", opts.enable)
+	return nil
+}
+
+func CopyProtectionBuilder() *cobra.Command {
+	opts := new(CopyProtectionOpts)
+	use := "copyProtection"
+	cmd := &cobra.Command{
+		Use:       use,
+		Aliases:   cli.GenerateAliases(use),
+		Args:      cobra.MatchAll(cobra.OnlyValidArgs, cobra.ExactArgs(1)),
+		ValidArgs: []string{enable, disable},
+		Short:     "Enable or disable copyprotection of the backup compliance policy for your project.",
+		PreRunE: func(cmd *cobra.Command, args []string) error {
+			// we throw an error if 1 argument is not present,
+			// so we can safely access args[0].
+			if args[0] == enable {
+				opts.enable = true
+			} else {
+				opts.enable = false
+			}
+			return opts.PreRunE(
+				opts.ValidateProjectID,
+				opts.initStore(cmd.Context()),
+				opts.InitOutput(cmd.OutOrStdout(), ""),
+				opts.PreRun,
+			)
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return opts.Run()
+		},
+	}
+
+	cmd.Flags().StringVar(&opts.ProjectID, flag.ProjectID, "", usage.ProjectID)
```

Comment:
[q] where are the other flags for this command? even if we don't want to add the happy flow, can we please make sure we're adding the planned command? 

```shell
  atlas backups compliancepolicy enable [flags]
Flags:
  --authorizedEmail string    Email address of a security or legal representative
  --projectId string          Hexadecimal string that identifies the project to use. This option overrides the settings in the configuration file or environment variable.
  --force                Indicates whether to skip the confirmation prompt before proceeding with the requested
```

Reply:
also, curious to know why we decide to cover happy flow in e2e task, can we please create another task for the happy flow as e2e already covers other things?

Reply:
> [q] where are the other flags for this command?


I believe you've accidentally mixed things up: this is the `copyprotection` command, not the `enable` command. Let me know if I understood you correctly. The `copyprotection` command has only 1 flag (`--projectId`), which is implemented.

>also, curious to know why we decide to cover happy flow in e2e task, can we please create another task for the happy flow as e2e already covers other things?

We can't test happy flows because we don't (yet) have any commands to enable BCP. I imagined the E2E task to uncomment one line in the describe test and add this happy flow test. It basically only exists because there was still some discussion around the enable & setup commands while I already started implementing. Given this context, would you still like me to create another task or is it fine to keep them combined?

Reply:
jumping here to mention this is missing output and this still doesn't work with `-o json`, please make sure that runing this command `-o json` will print a json of the reponse

Reply:
@gssbzn do we have some kind of policy that every command must support -o (or some other reason that requires the flag)? Asking because in the spec document, it was not planned to have `-o` for this command.

Reply:
all commands that hit an API should support O, is part of making the CLI friendly for programmatic users that want to script things

Reply:
Thanks